src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -0,0 +1,536 @@
+        saving: 'success',
+      });
+    } else {
+      this.snackbarText = 'Something went wrong while trying to save the new primary applicant.';
+      this.snackbarSeverity = 'error';
+      this.setState({ snackbarOpen: true, saving: 'failed' });
+    }
+    if (this.saveButtonAlertTimeout === null) {
+      // Shows the success icon for 6 seconds and then returns back to normal button
+      this.saveButtonAlertTimeout = setTimeout(() => {
+        this.saveButtonAlertTimeout = null;
+        this.setState({ saving: false });
+      }, 6000);
+    }
+  }
+
+  /**
+   * Callback for applicant list remove button
+   * @param {String} profileToRemove Username for student
+   */
+  handleRemove = (profileToRemove) => {
+    this.setState({ updating: true });
+    if (profileToRemove) {
+      let applicants = this.state.applicants; // make a separate copy of the array
+      let index = applicants.indexOf(profileToRemove);
+      if (index !== -1) {
+        applicants.splice(index, 1);
+        this.setState({ applicants });
+      }
+    }
+    this.setState({ updating: false });
+  };
+
+  /**
+   * Callback for apartment application save button
+   */
+  handleSaveButtonClick = () => {
+    let debugMessage = 'DEBUG: Save button was clicked'; //! DEBUG
+    console.log(debugMessage); //! DEBUG
+    // The method is separated from callback because the housing API service must be handled inside an async method
+    this.saveApplication(
+      this.state.applicationID,
+      this.state.primaryUsername,
+      this.state.applicants,
+    );
+  };
+
+  /**
+   * Save the current state of the application to the database
+   * @param {Number} applicationID the application ID number if it is known, else it is -1
+   * @param {String} primaryUsername the student username of the person filling out the application
+   * @param {StudentProfileInfo[]} applicants Array of StudentProfileInfo objects
+   */
+  async saveApplication(applicationID, primaryUsername, applicants) {
+    this.setState({ saving: true });
+    this.saveButtonAlertTimeout = null;
+    let result = null;
+    try {
+      result = await housing.saveApartmentApplication(applicationID, primaryUsername, applicants);
+    } catch {
+      result = false;
+    }
+    if (result !== null) {
+      console.log(result); //! DEBUG
+      this.setState({ applicationID: result, saving: 'success' });
+    } else {
+      this.snackbarText = 'Something went wrong while trying to save the application.';
+      this.snackbarSeverity = 'error';
+      this.setState({ snackbarOpen: true, saving: 'failed' });
+    }
+    if (this.saveButtonAlertTimeout === null) {
+      // Shows the success icon for 6 seconds and then returns back to normal button
+      this.saveButtonAlertTimeout = setTimeout(() => {
+        this.saveButtonAlertTimeout = null;
+        this.setState({ saving: false });
+      }, 6000);
+    }
+  }
+
+  handleCloseSnackbar = (event, reason) => {
```

Comment:
It might be better for snack bar to close with clickaways too.

Reply:
I did this so that it will either close after it times out or when the user deliberately clicks the close button in the snackbar. 
The snackbar is a small, non-invasive notification at the bottom of the screen, so it would very easy for someone to click anywhere else on the page and accidentally close the snackbar before they have a chance to read it (I am also speaking from personal experience too).